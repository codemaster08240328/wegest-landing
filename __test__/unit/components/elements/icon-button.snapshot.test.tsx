import renderer from 'react-test-renderer'
import IconButton from '@/components/elements/icon-button'

describe('<IconButton> snapshot', () => {
  it('renders correclty', () => {
    expect(
      renderer.create(<IconButton icon='add' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders with edge', () => {
    expect(
      renderer
        .create(
          <IconButton icon='add' edge='start'>
            Hello!
          </IconButton>,
        )
        .toJSON(),
    ).toMatchSnapshot()
  })

  it('renders with a class', () => {
    expect(
      renderer
        .create(<IconButton icon='add' className='class-test' />)
        .toJSON(),
    ).toMatchSnapshot()
  })
})
