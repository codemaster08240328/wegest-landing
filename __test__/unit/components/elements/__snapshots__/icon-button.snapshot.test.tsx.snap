// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton> snapshot renders correclty 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root Component-root-1 Component-root-2"
  data-testid="el-icon-button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root"
      data-testid="el-icon"
    >
      add
    </span>
  </span>
</button>
`;

exports[`<IconButton> snapshot renders with a class 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root Component-root-1 Component-root-4 class-test"
  data-testid="el-icon-button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root"
      data-testid="el-icon"
    >
      add
    </span>
  </span>
</button>
`;

exports[`<IconButton> snapshot renders with edge 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root Component-root-1 Component-root-3"
  data-testid="el-icon-button"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root"
      data-testid="el-icon"
    >
      add
    </span>
    Hello!
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
